File: packages/js-dpp/lib/errors/SerializedObjectParsingError.js
const ConsensusError = require('./ConsensusError');

class SerializedObjectParsingError extends ConsensusError {
  /**
   * @param {Buffer|string} payload
   * @param {Error} parsingError
   */
  constructor(payload, parsingError) {
    super(
      `Parsing of a serialized object failed due to: ${parsingError.message}`,
    );

    this.payload = payload;
    this.parsingError = parsingError;
  }

  /**
   * Get object payload
   *
   * @return {Buffer|string}
   */
  getPayload() {
    return this.payload;
  }

  /**
   * Get parsing error
   *
   * @return {Error}
   */
  getParsingError() {
    return this.parsingError;
  }
}

module.exports = SerializedObjectParsingError;

File: packages/rs-dpp/src/errors/consensus/basic/decode/serialized_object_parsing_error.rs
use crate::consensus::basic::BasicError;
use crate::consensus::ConsensusError;
use crate::errors::ProtocolError;
use bincode::{Decode, Encode};
use platform_serialization_derive::{PlatformDeserialize, PlatformSerialize};
use thiserror::Error;

#[derive(
    Error, Debug, Clone, PartialEq, Eq, Encode, Decode, PlatformSerialize, PlatformDeserialize,
)]
#[error("Parsing of serialized object failed due to: {parsing_error}")]
#[platform_serialize(unversioned)]
pub struct SerializedObjectParsingError {
    /*

    DO NOT CHANGE ORDER OF FIELDS WITHOUT INTRODUCING OF NEW VERSION

    */
    parsing_error: String,
}

impl SerializedObjectParsingError {
    pub fn new(parsing_error: String) -> Self {
        Self { parsing_error }
    }

    pub fn parsing_error(&self) -> &str {
        &self.parsing_error
    }
}
impl From<SerializedObjectParsingError> for ConsensusError {
    fn from(err: SerializedObjectParsingError) -> Self {
        Self::BasicError(BasicError::SerializedObjectParsingError(err))
    }
}
